contracts/core/Swapica.sol

```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+
+import "@dlsl/dev-modules/libs/arrays/Paginator.sol";
+
+import "../interfaces/core/ISwapica.sol";
+
+import "./Globals.sol";
+import "../multisig/Signers.sol";
+
+contract Swapica is ISwapica, UUPSUpgradeable, Signers {
+    using Paginator for uint256[];
+    using Math for uint256;
+    using SafeERC20 for IERC20;
+
+    Order[] internal _orders;
+    Match[] internal _matches;
+
+    mapping(address => User) internal _userInfos;
+
+    event OrderUpdated(uint256 indexed orderId, OrderStatus status);
+    event MatchUpdated(uint256 indexed matchId, State status);
```

Comment:
We should probably have 2 more events: OrderCreated, MatchCreated

Reply:
done